win32ss/drivers/videoprt/int10.c

```diff
@@ -23,11 +23,61 @@
 
 #include <ndk/kefuncs.h>
 #include <ndk/halfuncs.h>
+#include <ndk/mmfuncs.h>
 
 #define NDEBUG
 #include <debug.h>
 
 /* PRIVATE FUNCTIONS **********************************************************/
+#define IsLowV86Mem(_Seg, _Off) ((((_Seg) << 4) + (_Off)) < (0xa0000))
+
+/* Those two functions below are there so that CSRSS can't access low mem.
+ * Expecially, MAKE IT CRASH ON NULL ACCESS */
+static
+VOID
+ProtectLowV86Mem(VOID)
+{
+    /* We pass a non-NULL address so that ZwAllocateVirtualMemory really does it
+     * And we truncate one page to get the right range spanned. */
+    PVOID BaseAddress = (PVOID)1;
+    NTSTATUS Status;
+    SIZE_T ViewSize = 0xa0000 - PAGE_SIZE;
+
+    /* We should only do that for CSRSS. */
+    ASSERT(PsGetCurrentProcess() == (PEPROCESS)CsrProcess);
+
+    /* Commit (again) the pages, but with PAGE_NOACCESS protection */
+    Status = ZwAllocateVirtualMemory(NtCurrentProcess(),
+                                     &BaseAddress,
+                                     0,
+                                     &ViewSize,
+                                     MEM_COMMIT,
+                                     PAGE_NOACCESS);
+    ASSERT(NT_SUCCESS(Status));
+}
+
+static
```

Comment:
These two functions are so similar that you could merge them into one, and add a BOOLEAN parameter that decides whether the function will protect (use `PAGE_NOACCESS`) or unprotect (use `PAGE_READWRITE`) the memory.

Reply:
I hate it when you have a function like this:

`DisableOrEnableThis(BOOL` IsThisParamTrueForEnableOrDisableIDKDudeWTF)

And those two are small enough to keep them separate and make clear to anyone what the workflow is.

Reply:
> IsThisParamTrueForEnableOrDisableIDKDudeWTF

That's why you can write code documentation too.